generator/api/src/main/java/io/fabric8/maven/generator/api/FromSelector.java

```diff
@@ -126,7 +129,13 @@ public boolean isJava11() {
     }
 
     private boolean isEleven(String s) {
-        return Configs.asInt(s) == 11;
+        boolean result = false;
+        try {
+            result = Configs.asInt(s) == 11;
+        } catch (NumberFormatException e) {
+        } finally {
```

Comment:
Is it okay to have an empty catch block? Can we throw this exception instead rather than swallowing it?

Reply:
If we throw the exception we won't be able to handle Java versions formatted as 1.x ...

Reply:
This catch block is to handle for that..its a necessity

Reply:
Can we at least log something in that block rather than leaving it empty? Empty catch blocks are bad practice and make debugging difficult

Reply:
https://github.com/fabric8io/fabric8-maven-plugin/pull/1758/commits/c31aa602f2d78470130abe2305e67d14b5efac28

^this much alteration is required to add a logger here. Worth it?

This try-catch block is there for developer's aid and not users' IMO... if a user provides a 1.x formatted release version, then I don't think he needs to be informed about that in logs, as it is a normal practice.

A comment explaining the developer's intention with the try-catch block should have been enough I think but I did it anyway... you guys decide if you want to merge with the logger or just a comment..I'll update the PR accordingly. @rohanKanojia @manusa

Reply:
Also, the log will be invoked for every generator :/ 

```
[INFO] -------< io.fabric8:fabric8-maven-sample-spring-boot-with-yaml >--------
[INFO] Building Fabric8 Maven :: Sample :: Spring Boot Web 4.4-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- fabric8-maven-plugin:4.4-SNAPSHOT:resource (default-cli) @ fabric8-maven-sample-spring-boot-with-yaml ---
[INFO] F8: Using Container image name of namespace: spring-boot-ns
[INFO] F8: spring-boot: 1.8 target Java version detected
[INFO] F8: openliberty: 1.8 target Java version detected
[INFO] F8: wildfly-swarm: 1.8 target Java version detected
[INFO] F8: java-exec: 1.8 target Java version detected
[INFO] F8: webapp: 1.8 target Java version detected
[INFO] F8: vertx: 1.8 target Java version detected
[INFO] F8: quarkus: 1.8 target Java version detected
[INFO] F8: thorntail-v2: 1.8 target Java version detected
[INFO] F8: karaf: 1.8 target Java version detected
[INFO] F8: Running generator spring-boot
[INFO] F8: spring-boot: Using Container image fabric8/s2i-java:2.3 as base / builder
[INFO] F8: using resource templates from /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/src/main/fabric8
[INFO] F8: fmp-namespace: Adding a default Namespace:spring-boot-ns
[INFO] F8: f8-healthcheck-spring-boot: Adding readiness probe on port 8080, path='/actuator/health', scheme='HTTP', with initial delay 10 seconds
[INFO] F8: f8-healthcheck-spring-boot: Adding liveness probe on port 8080, path='/actuator/health', scheme='HTTP', with initial delay 180 seconds
[INFO] F8: fmp-revision-history: Adding revision history limit to 2
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/kubernetes/spring-boot-ns-namespace.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/kubernetes/fabric8-maven-sample-spring-boot-with-yaml-template.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/kubernetes/fabric8-maven-sample-spring-boot-with-yaml-service.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/kubernetes/password-secret.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/kubernetes/fabric8-maven-sample-spring-boot-with-yaml-deployment.yml resource
[INFO] F8: using resource templates from /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/src/main/fabric8
[INFO] F8: fmp-namespace: Adding a default Projectspring-boot-ns
[INFO] F8: f8-healthcheck-spring-boot: Adding readiness probe on port 8080, path='/actuator/health', scheme='HTTP', with initial delay 10 seconds
[INFO] F8: f8-healthcheck-spring-boot: Adding liveness probe on port 8080, path='/actuator/health', scheme='HTTP', with initial delay 180 seconds
[INFO] F8: fmp-revision-history: Adding revision history limit to 2
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/fabric8-maven-sample-spring-boot-with-yaml-template.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/fabric8-maven-sample-spring-boot-with-yaml-service.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/password-secret.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/fabric8-maven-sample-spring-boot-with-yaml-deploymentconfig.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/spring-boot-ns-project.yml resource
[INFO] F8: validating /home/dgaur/dev-gaur/fabric8io/fabric8-maven-plugin/samples/spring-boot-with-yaml/target/classes/META-INF/fabric8/openshift/fabric8-maven-sample-spring-boot-with-yaml-route.yml resource
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14.507 s
[INFO] Finished at: 2019-11-15T04:38:25+05:30
[INFO] ------------------------------------------------------------------------

```